vendor/github.com/openshift/library-go/pkg/monitor/health/prober.go

```diff
@@ -0,0 +1,377 @@
+package health
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/transport"
+	"k8s.io/klog/v2"
+)
+
+var (
+	defaultProbeResponseTimeout = 1 * time.Second
+	defaultProbeInterval        = 2 * time.Second
```

Comment:
these look too aggressive, there is always a trade off, but being agressive can lead to false negative (i.e. a probe that takes more than 1 second in a busy environment should not be rare)

Reply:
these are the default values, for KCM we set https://github.com/openshift/kubernetes/pull/603/commits/f26ba46ac7fc2e1198c8f5327fef3dd93d6ba8f5#diff-0eb83e2c7853c63513dc9b1bcf43e9a4d11778eccb562c51d30c8379025159d0R43